src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -59,15 +65,20 @@ def _builtin_op(*fields: common.Field | core_defs.Scalar) -> NdArrayField:
                 if f.domain == domain_intersection:
                     transformed.append(xp.asarray(f.ndarray))
                 else:
-                    f_broadcasted = _broadcast(f, domain_intersection.dims)
+                    f_broadcasted = (
+                        _broadcast(f, domain_intersection.dims)
+                        if f.domain.dims != domain_intersection.dims
+                        else f
```

Comment:
```suggestion
                        _broadcast(f, domain_intersection.dims)
                        if f.domain.dims != domain_intersection.dims
                        else f
```
I would expect this to be part of the broadcast operation. Can't we put it there?

Reply:
actually not sure why I cared for this optimization, done.